src/tests/Microsoft.PowerFx.Interpreter.Tests/RecalcEngineTests.cs

```diff
@@ -335,6 +335,51 @@ public void DefFunc()
             Assert.Equal(17.0, recalcEngine.Eval("foo(Float(3),Float(4)) + 5").ToObject());
         }
 
+        [Fact]
+        public void DefTypeAndFunc()
+        {
+            var config = new PowerFxConfig();
+            var recalcEngine = new RecalcEngine(config);
+            var parserOptions = new ParserOptions()
+            {
+                AllowsSideEffects = false,
+            };
+            recalcEngine.DefineType("Person = Type({ Age: Number});", parserOptions);
+            IEnumerable<ExpressionError> enumberable = recalcEngine.DefineFunctions("getAge(p: Person): Number = p.Age;").Errors;
+            Assert.False(enumberable.Any());
+            Assert.Equal(1.0, recalcEngine.Eval("getAge({Age: Float(1)})").ToObject());
+        }
+
+        [Fact]
+        public void DefComplexTypeWithFunc()
+        {
+            var config = new PowerFxConfig();
+            var recalcEngine = new RecalcEngine(config);
+            var parserOptions = new ParserOptions()
+            {
+                AllowsSideEffects = false,
+            };
+            recalcEngine.DefineType("Complex = Type({ A: {B: Number}});", parserOptions);
+            IEnumerable<ExpressionError> enumberable = recalcEngine.DefineFunctions("foo(p: Complex): Number = p.A.B;").Errors;
+            Assert.False(enumberable.Any());
+            Assert.Equal(1.0, recalcEngine.Eval("foo({A: {B: Float(1.0)}})").ToObject());
+        }
+
+        [Fact]
+        public void DefManyType()
+        {
+            var config = new PowerFxConfig();
+            var recalcEngine = new RecalcEngine(config);
+            var parserOptions = new ParserOptions()
+            {
+                AllowsSideEffects = false,
+            };
+            recalcEngine.DefineType("A = Type({ num: Number}); B = Type({ a: A }); C = Type({ b: B, a: A});", parserOptions);

```

Comment:
>("A = Type({ num: Number}); B = Type({ a: A }); C = Type({ b: B, a: A}); [](http://example.com/codeflow?start=35&length=72)

Can you add a negative case for self-referencing type definition?
For example, A = Type({num:A})

Reply:
Added a negative test case, but no negative test case for self-referencing type definition